src/renderer/qt/qt_window_manager.cc

```diff
@@ -341,16 +341,61 @@ void FillCandidates(const commands::Candidates &candidates,
 }
 }  // namespace
 
+struct VirtualRect {
+  const QScreen *screen;
+  const Rect rect;
+
+  VirtualRect(const QScreen *screen, const Rect rect)
+      : screen(screen), rect(rect) {}
+};
+
+Rect TranslateToVirtualRectWithScreen(const QScreen *screen,
+                                      const Rect &native_rect) {
+  const double device_pixel_ratio = screen->devicePixelRatio();
+  // screen_left, screen_top have the save value in both virtual and native
+  // coordinate
+  const int screen_left = screen->geometry().x();
+  const int screen_top = screen->geometry().y();
+  const int vx =
+      (native_rect.Left() - screen_left) / device_pixel_ratio + screen_left;
+  const int vy =
+      (native_rect.Top() - screen_top) / device_pixel_ratio + screen_top;
+
+  return Rect(vx, vy, native_rect.Width() / device_pixel_ratio,
+              native_rect.Height() / device_pixel_ratio);
+}
+
+VirtualRect TranslateToVirtualRect(const Rect &native_rect) {
```

Comment:
optional: I prefer to have this function as a static member of `VirtualRect`, but it's a personal preference.

e.g. `static VirtualRect FromNativeRect(const &native_rect)`

Reply:
I made a factory method `FromNativeRect` and moved it to `VirtualRect`